### PR TITLE
Update polls to indicate what option was voted for

### PR DESCRIPTION
**Pull request by bumbadadabum**

### CODE REVIEW

**Comment by DanUgore** [comment]
rude

**Comment by bumbadadabum** [comment]
I can change the branch into something more serious if that's really needed.

File: chat-plugins/poll.js [hunk 1]
```diff
@@ -78,20 +80,17 @@ var Poll = (function () {
...
 
 	Poll.prototype.update = function () {
-		var results = this.generateResults();
-
 		// Update the poll results for everyone that has voted
 		for (var i in this.room.users) {
 			var user = this.room.users[i];
 			if (this.voters.has(user.latestIp)) {
-				user.sendTo(this.room, '|uhtmlchange|poll' + this.room.pollNumber +  '|' + results);
+				user.sendTo(this.room, '|uhtmlchange|poll' + this.room.pollNumber +  '|' + this.generateResults(false, user));
```
Comment (Zarel) [inline chat-plugins/poll.js]: Moving `generateResults` inside an inner loop sounds laggy. There are ways to do it that aren't laggy. Who's the one who can't code again? ;)

Comment (bumbadadabum) [inline chat-plugins/poll.js]: Yeah there are better ways will try to implement that soon

**Comment by bumbadadabum** [comment]
Tried to prove I can code

**Comment by Slayer95** [comment]
Options chosen should probably be mapped from users, rather than IPs.
Otherwise, it will display funny results sometimes.

**Comment by Zarel** [comment]
You'll need to rebase using `let` now. :P

**Comment by bumbadadabum** [comment]
@Slayer95 How would that give funny results? Everything else is also mapped to ip.

**Comment by Slayer95** [comment]
Everything else is fine mapped to IP.
If users are as well, then people with dynamic IPs will see their choices refreshed / removed / void / nulled / tinkered with.

**Comment by bumbadadabum** [comment]
They would anyway.
Op 7 nov. 2015 17:25 schreef "Slayer95" <notifications@github.com>:

**Comment by bumbadadabum** [comment]
Added a commit that does that. With this, a user would need a different userid and IP to be able to vote again, which is one step away from ban evading to vote twice in a poll.

**Comment by Zarel** [comment]
> The only way to fix this would be to add a second list with usernames and choices as well. Is that worth it for a very small issue?

I would say no. IPs do not change in a single session. "Dynamic" IPs aren't _that_ dynamic.

**Comment by Slayer95** [comment]
Thing is, you don't need to map IPs to anything. We just keep track of the IPs and prevent duplications, since we don't allow switching votes, and I don't see any reason to support it at any point.

**Comment by bumbadadabum** [comment]
I find IP a better way to check who a person is than username.

**Comment by Slayer95** [comment]
That's an unfounded statement as this is not moderating, and inaccurate. What does "better" mean? Less false positives? Less false negatives? My proposal adds no more false negatives in the critical stage (voting), and decreases both false positives and false negatives in the feedback.

**Comment by bumbadadabum** [comment]
Yes, but if I go on an alt I don't want to see that I didn't vote.

**Comment by Zarel** [comment]
My perspective, which hasn't changed: IP isn't perfect but the messiness to accurately keep track of the use through IP changes isn't worth it imo

Just using pure IPs makes for simple code and is good enough

**Comment by Slayer95** [comment]
So are we willing to leak information just so that people can easily switch alts?
We support alt switching way too much... That's why every once in a while someone says "I disconnected and my account got stuck and I can't go back to my battle, plz help!!"...

**Comment by Zarel** [comment]
What information is being leaked? That's still unclear to me.

The reason they can't go back to their battle is because Cathy got rid of some session code I'd made :(

**Comment by bumbadadabum** [comment]
Does anyone else have suggestions on how to improve this?

**Comment by Zarel** [comment]
If you have a `const permission`, you might as well just make your own permission and put it in `config-example`.

**Pull request merged**